### PR TITLE
Swagger docs

### PR DESCRIPTION
**Pull request by emmetthitz**
Annotations done. Would be good to get eyes on it to see if I missed something.

- [x] Fix integration test endpoints
- [x] Clean up some messes
- [x] Figure out why spec generation doesn't work
- [x] Embed swagger-ui

### CODE REVIEW

File: pkg/server/http/handler/clusters.go [hunk 1]
```diff
@@ -211,6 +269,7 @@ func (api Clusters) GetAll(w http.ResponseWriter, r *http.Request, p httprouter.
 		return
 	}
 	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
```
Comment (emmetthitz) [inline pkg/server/http/handler/clusters.go]: Woops

File: Makefile [hunk 2]
```diff
@@ -17,7 +17,7 @@ GLIDE_VERSION = v0.13.1
 ANSIBLE_VERSION = 2.3.0.0
 TERRAFORM_VERSION = 0.11.0
 KUBERANG_VERSION = v1.2.2
-GO_VERSION = 1.9.2
+GO_VERSION = 1.9.3
```
Comment (dkoshkin) [inline Makefile]: Is this required, it might be better to have this in a different PR. Also there was a security patch release with Go 1.9.4.

File: circle.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ jobs:
       - checkout
       - run: 
           name: Install unzip
-          command: apt update && apt install unzip
+          command: sudo apt update && sudo apt install unzip
```
Comment (dkoshkin) [inline circle.yml]: Was this not working before?

File: pkg/server/http/handler/clusters.go [hunk 4]
```diff
@@ -32,35 +32,93 @@ type Clusters struct {
...
+	//
+	// in: body
+	// required: true
+	Provisioner store.Provisioner `json:"provisioner"`
 }
 
 // The Provisioner defines the infrastructure provisioner that should be used
 // for hosting the cluster
 type sanitizedProvisioner struct {
+	// Options: aws, azure, digital ocean
```
Comment (dkoshkin) [inline pkg/server/http/handler/clusters.go]: + any other provider user defined.

File: pkg/server/http/server.go [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
+//     BasePath: /api/v1
+//     Version: 0.0.1
+//     License: Apache 2.0 http://www.apache.org/licenses/
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//     - application/x-yaml
```
Comment (dkoshkin) [inline pkg/server/http/server.go]: I still don't know how I feel about using this instead of `text/plain`

File: circle.yml (hunk 3 above)
Comment (emmetthitz) [reply circle.yml]: It's related to the previous comment. The updated container doesn't give root by default.

File: Makefile (hunk 2 above)
Comment (emmetthitz) [reply Makefile]: Yeah, there's a handful of things that got tracked with this that I added by accident.

File: pkg/server/http/server.go (hunk 5 above)
Comment (emmetthitz) [reply pkg/server/http/server.go]: The more I think about it, the more I think it's kind of misleading. I think plain text generally implies that there's no structure to it. I.E., a log file or some other kind of typical human writing.

File: integration-tests/daemon_test.go [hunk 6]
```diff
@@ -95,7 +96,7 @@ The error: %v
 				pb, err := json.Marshal(payload)
 				Expect(err).ToNot(HaveOccurred())
 
-				clustersEndpoint := fmt.Sprintf("https://localhost:%d/clusters", daemonPort)
+				clustersEndpoint := fmt.Sprintf("https://localhost:%d%s/clusters", daemonPort, ketHttp.APIPath)
```
Comment (dkoshkin) [inline integration-tests/daemon_test.go]: Looks like missing a slash between `%d%s`

Comment (emmetthitz) [reply integration-tests/daemon_test.go]: APIPath begins with a `/`. I'll change that as well if you think it makes more sense stylistically.

Comment (dkoshkin) [reply integration-tests/daemon_test.go]: Its inconsistent with [this](https://github.com/apprenda/kismatic/pull/1102/files#diff-334a3bd3e1b27ff451cbce986a0b9623R144) so one of them will be wrong, I do think its should have `/`

Comment (emmetthitz) [reply integration-tests/daemon_test.go]: `httprouter` complains if `APIPath` doesn't begin with a `/`. I corrected the other version instead of changing this.
